web/src/main/java/com/bazaarvoice/emodb/web/resources/blob/BlobStoreResource1.java

```diff
@@ -109,13 +108,11 @@ public BlobStoreResource1(BlobStore blobStore, DataCenters dataCenters, Set<Stri
     public Iterator<Table> listTables(@QueryParam("from") final String fromKeyExclusive,
                                       @QueryParam("limit") @DefaultValue("10") LongParam limit,
                                       final @Authenticated Subject subject) {
-        final Iterator<Table> tables = _blobStore.listTables(Strings.emptyToNull(fromKeyExclusive), limit.get());
-        final UnmodifiableIterator<Table> permittedTables = Iterators.filter(tables, new Predicate<Table>() {
-            @Override public boolean apply(final Table input) {
-                return subject.hasPermission(Permissions.readBlobTable(new NamedResource(input.getName())));
-            }
-        });
-        return streamingIterator(permittedTables);
+        final String fromTableExclusive = Strings.emptyToNull(fromKeyExclusive);
+        final Iterator<Table> lazyAllTablesPaged = new ListTablesPager(_blobStore, fromTableExclusive, limit.get());
+        final Iterator<Table> permittedTables = Iterators.filter(lazyAllTablesPaged, input -> subject.hasPermission(Permissions.readBlobTable(new NamedResource(input.getName()))));
```

Comment:
Eh, I think I think the current breakdown better. If you feel strongly about it, or if this is problematic for some reason, we can coalesce them.